pycbc/coordinates/space.py

```diff
@@ -0,0 +1,906 @@
+        The ecliptic latitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+
+    Returns
+    -------
+    t_ssb : float
+        The time when a GW signal arrives at the origin of SSB frame.
+    """
+    k = localization_to_propagation_vector(longitude_ssb, latitude_ssb)
+    # Earth is moving, when GW arrives at Earth center,
+    # time is t_geo, not t_ssb.
+    p = earth_position_ssb(t_geo)[0]
+
+    def equation(t_ssb):
+        return t_geo - t_ssb - np.vdot(k, p) / c.value
+
+    return fsolve(equation, t_geo)[0]
+
+
+def ssb_to_geo(t_ssb, longitude_ssb, latitude_ssb, polarization_ssb,
+               use_astropy=True):
+    """ Converting the arrive time, the sky localization, and the polarization
+    from the SSB frame to the geocentric frame.
+
+    Parameters
+    ----------
+    t_ssb : float or numpy.array
+        The time when a GW signal arrives at the origin of SSB frame.
+        In the unit of 's'.
+    longitude_ssb : float or numpy.array
+        The ecliptic longitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    latitude_ssb : float or numpy.array
+        The ecliptic latitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    polarization_ssb : float or numpy.array
+        The polarization angle of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    use_astropy : bool
+        Using Astropy to calculate the sky localization or not.
+        Default is True.
+
+    Returns
+    -------
+    (t_geo, longitude_geo, latitude_geo, polarization_geo) : tuple
+    t_geo : float or numpy.array
+        The time when a GW signal arrives at the origin of geocentric frame.
+        In the unit of 's'.
+    longitude_geo : float or numpy.array
+        The longitude of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    latitude_geo : float or numpy.array
+        The latitude of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    polarization_geo : float or numpy.array
+        The polarization angle of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    """
+    if isinstance(t_ssb, np.ndarray):
+        t_ssb_array = t_ssb.copy()
+    else:
+        t_ssb_array = np.array([t_ssb])
+    if isinstance(longitude_ssb, np.ndarray):
+        longitude_ssb_array = longitude_ssb.copy()
+    else:
+        longitude_ssb_array = np.array([longitude_ssb])
+    if isinstance(latitude_ssb, np.ndarray):
+        latitude_ssb_array = latitude_ssb.copy()
+    else:
+        latitude_ssb_array = np.array([latitude_ssb])
+    if isinstance(polarization_ssb, np.ndarray):
+        polarization_ssb_array = polarization_ssb.copy()
+    else:
+        polarization_ssb_array = np.array([polarization_ssb])
+    num = len(t_ssb_array)
+    t_geo_array, longitude_geo_array = np.zeros(num), np.zeros(num)
+    latitude_geo_array, polarization_geo_array = np.zeros(num), np.zeros(num)
+
+    for i in range(num):
```

Comment:
Why are you looping over num here rather than just doing vector / ndarray operations as appropriate? This would be a much slower implementation if you want to calculate for many points.

Reply:
@WuShichao This issue actually still needs to be addressed.